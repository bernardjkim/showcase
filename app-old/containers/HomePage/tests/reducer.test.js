import homePageReducer, { initialState } from '../reducer';
import {
  clearState,
  loadArticlesAll,
  loadArticlesAllError,
  loadArticlesAllSuccess,
  loadNext,
  loadNextError,
  loadNextSuccess,
} from '../actions';

describe('homePageReducer', () => {
  let state;
  beforeEach(() => {
    state = initialState;
  });

  it('returns the initial state', () => {
    expect(homePageReducer(state, {})).toMatchSnapshot();
  });
  it('handle the clear state action', () => {
    expect(homePageReducer(state, clearState())).toMatchSnapshot();
  });

  it('handle the load articles all action', () => {
    expect(homePageReducer(state, loadArticlesAll())).toMatchSnapshot();
  });

  it('handle the load articles all success action', () => {
    const articles = [];
    expect(
      homePageReducer(state, loadArticlesAllSuccess(articles)),
    ).toMatchSnapshot();
  });

  it('handle the load articles all error action', () => {
    const error = new Error('Test Error');
    expect(
      homePageReducer(state, loadArticlesAllError(error)),
    ).toMatchSnapshot();
  });
  it('handle the load next action', () => {
    expect(homePageReducer(state, loadNext())).toMatchSnapshot();
  });

  it('handle the load next success action', () => {
    const articles = [];
    expect(homePageReducer(state, loadNextSuccess(articles))).toMatchSnapshot();
  });

  it('handle the load articles error action', () => {
    const error = new Error('Test Error');
    expect(homePageReducer(state, loadNextError(error))).toMatchSnapshot();
  });
});
