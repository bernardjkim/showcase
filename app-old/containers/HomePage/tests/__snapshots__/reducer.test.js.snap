// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homePageReducer handle the clear state action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "articles": Immutable.List [],
  "offset": 0,
}
`;

exports[`homePageReducer handle the load articles all action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
  "articles": Immutable.List [],
  "offset": 0,
}
`;

exports[`homePageReducer handle the load articles all error action 1`] = `
Immutable.Map {
  "loading": false,
  "error": [Error: Test Error],
  "articles": Immutable.List [],
  "offset": 0,
}
`;

exports[`homePageReducer handle the load articles all success action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "articles": Immutable.List [],
  "offset": 0,
}
`;

exports[`homePageReducer handle the load articles error action 1`] = `
Immutable.Map {
  "loading": false,
  "error": [Error: Test Error],
  "articles": Immutable.List [],
  "offset": 0,
}
`;

exports[`homePageReducer handle the load next action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
  "articles": Immutable.List [],
  "offset": 0,
}
`;

exports[`homePageReducer handle the load next success action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "articles": Immutable.List [],
  "offset": 0,
}
`;

exports[`homePageReducer returns the initial state 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "articles": Immutable.List [],
  "offset": 0,
}
`;
